legacy_packages/sdk/src/evm/common/any-evm-utils/fetchAndCacheDeployMetadata.ts

```diff
@@ -22,14 +23,24 @@ const deployMetadataCache =
 export async function fetchAndCacheDeployMetadata(
   publishMetadataUri: string,
   storage: ThirdwebStorage,
+  compilerOptions?: CompilerOptions,
 ): Promise<DeployMetadata> {
-  const cached = deployMetadataCache.get(publishMetadataUri);
+  let compiler = compilerOptions
+    ? `${compilerOptions.compilerType}_${
+        compilerOptions.compilerVersion || ""
+      }_${compilerOptions.evmVersion || ""}
+  `
+    : "default";
+  const cacheKey = `${compiler} - ${publishMetadataUri}`;
```

Comment:
```suggestion
  const cacheKey = `${compiler}_${publishMetadataUri}`;
```

consistent with the `compiler` part of the key above